target/linux/rockchip/image/Makefile

```diff
@@ -0,0 +1,49 @@
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/image.mk
+
+FAT32_BLOCK_SIZE=1024
+FAT32_BLOCKS=$(shell echo $$(($(CONFIG_TARGET_KERNEL_PARTSIZE)*1024*1024/$(FAT32_BLOCK_SIZE))))
+
+define Build/Compile
+	$(CP) $(LINUX_DIR)/COPYING $(KDIR)/COPYING.linux
+endef
+
+### Image scripts ###
+define Build/boot-common
+	# This creates a new FAT32 partition as an image file and 
+	# uses mtools to copy the dtb (as rockchip.dtb) and the kernel image (as kernel.img)
+	rm -f $@.boot
+	mkfs.fat -C $@.boot $(FAT32_BLOCKS)
+	mcopy -i $@.boot $(DTS_DIR)/$(DEVICE_DTS).dtb ::/rockchip.dtb
+	mcopy -i $@.boot $(IMAGE_KERNEL) ::/kernel.img
+endef
+
+define Build/boot-script
+	# Make an U-boot image and copy it to the FAT32 partition
+	mkimage -A arm -O linux -T script -C none -a 0 -e 0 -d mmc.bootscript $@-boot.scr
+	mcopy -i $@.boot $@-boot.scr ::/boot.scr
+endef
+
+define Build/pine64-img
+	# gen_pine64_img creates the final SD/eMMC images, 
+	# combining boot partition, root partition as well as the u-boot bootloader
+	./gen_pine64_img.sh $@ $@.boot $(IMAGE_ROOTFS) \
+		$(CONFIG_TARGET_KERNEL_PARTSIZE) $(CONFIG_TARGET_ROOTFS_PARTSIZE) \
+		$(STAGING_DIR_IMAGE) $(UBOOT_DEVICE_NAME) $(IMG_PART_SIGNATURE)
+endef
+
+### Devices ###
+define Device/Default
+	KERNEL := kernel-bin
+	IMAGES := sysupgrade.img.gz
```

Comment:
Add
`  SUPPORTED_DEVICES := $(subst _,$(comma),$(1))`
here and remove SUPPORTED_DEVICES for the actual device.

Reply:
Good idea, added, works!